### DISCUSSION TITLE
How can I diagnose high CPU usage in Cinnamon?

### DISCUSSION THREAD

**Discussion by CapnKernel** [linuxmint/discussions#238] [Questions]
I'm a very happy Cinnamon user since Gnome 2 days.  I use the Cinnamon that comes with Fedora (5.8.4).

Sometimes for some reason, the `cinnamon` process starts grabbing lots of CPU (130-180%), and I don't know why:

![image](https://github.com/linuxmint/discussions/assets/527149/7b2c5072-b265-4028-92cd-7ebc0062db7e)

How can I diagnose what cinnamon is doing when the CPU load is high?  Is there a profiler I can turn on or use?

Thank you.

**Comment by 100savage** [comment]
You might have a resource leak, press CTRL+Alt+Esc together to reset the wayward cinnamon process. Doing so won't close anything you have open but should help restore some normality.

**Comment by Acheron17** [comment]
Just trying to help here.  What is your CPU?

**Comment by Jopp-gh** [comment]
I agree with 100savage.
Also, not sure if related or not, try to check your cinnamon plugins, desklets and applets. Try to deactivate them one by one and see what happens

**Comment by CapnKernel** [comment]
Thank you, this problem has gone away.  It could be because I upgraded Cinnamon, or because I've been able to switch to accelerated mode rather than fallback mode.
